python-package/xgboost/core.py

```diff
@@ -176,11 +176,16 @@ def _maybe_pandas_data(data, feature_names, feature_types):
 
     data_dtypes = data.dtypes
     if not all(dtype.name in PANDAS_DTYPE_MAPPER for dtype in data_dtypes):
+        bad_types = [data_dtypes[i].name for i, dtype in
```

Comment:
Bug, doesn't output correct list of column names and instead would output actual data.